app/controllers/application_controller.rb

```diff
@@ -12,4 +13,43 @@ def login_required
   def guest_only
     redirect_to dashboard_path if logged_in?
   end
+
+  def obfuscate(hash)
+    require "base64"
+
+    hash.each do |k, v|
+      @encoded_key = Base64.encode64(k.to_s)
+      @encoded_value = Base64.encode64(v.to_s)
+    end
+
+    @encoded_key.delete!("\n")
+    @encoded_value.delete!("\n")
+
+    { @encoded_key => @encoded_value }
+  end
+
+  def deobfuscate(hash)
+    require "base64"
+
+    hash.each do |k, v|
+      @decoded_key = Base64.decode64(k.to_s)
+      @decoded_value = Base64.decode64(v.to_s)
+    end
+
+    @decoded_key.delete!("\n")
+    @decoded_value.delete!("\n")
+
+    { @decoded_key => @decoded_value }
+  end
+
+  def show_notification_count
```

Comment:
This logic belongs to the `TaskManagement` model, since we are basically making a query to the DB. What do you think?